plugins/modules/aci_aaa_role.py

```diff
@@ -0,0 +1,384 @@
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec
+
+PRIVILEGES = [
+    "admin",
+    "aaa",
+    "tenant-connectivity",
+    "tenant-protocol",
+    "vmm-policy",
+    "tenant-ext-connectivity",
+    "tenant-ext-protocol",
+    "tenant-qos",
+    "tenant-security",
+    "tenant-network-profile",
+    "tenant-epg",
+    "fabric-connectivity",
+    "fabric-protocol",
+    "fabric-equipment",
+    "access-connectivity",
+    "access-protocol",
+    "access-equipment",
+    "access-qos",
+    "nw-svc-params",
+    "ops",
+    "nw-svc-policy",
+    "site-admin",
+    "site-policy",
+    "config-manager",
+    "custom-privilege-1",
+    "custom-privilege-2",
+    "custom-privilege-3",
+    "custom-privilege-4",
+    "custom-privilege-5",
+    "custom-privilege-6",
+    "custom-privilege-7",
+    "custom-privilege-8",
+    "custom-privilege-9",
+    "custom-privilege-10",
+    "custom-privilege-11",
+    "custom-privilege-12",
+    "custom-privilege-13",
+    "custom-privilege-14",
+    "custom-privilege-15",
+    "custom-privilege-16",
+    "custom-privilege-17",
+    "custom-privilege-18",
+    "custom-privilege-19",
+    "custom-privilege-20",
+    "custom-privilege-21",
+    "custom-privilege-22",
+    "custom-port-privilege",
+]
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        name=dict(type="str", aliases=["aaa_role"]),
+        priv=dict(type="list", elements="str", choices=PRIVILEGES),
+        description=dict(type="str", aliases=["descr"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        name_alias=dict(type="str"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["name"]],
+            ["state", "present", ["name", "priv"]],
+        ],
+    )
+
+    name = module.params.get("name")
+    description = module.params.get("description")
+    priv = module.params.get("priv")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+
+    aci = ACIModule(module)
+    aci.construct_url(
+        root_class=dict(
+            aci_class="aaaRole",
+            aci_rn="userext/role-{0}".format(name),
+            module_object=name,
+            target_filter={"name": name},
+        ),
+    )
+    aci.get_existing()
+
+    if isinstance(priv, list):
+        formatted_privileges = ",".join(priv)
```

Comment:
if formatted_privileges only used in present state, do we want to move it into if state=="present"